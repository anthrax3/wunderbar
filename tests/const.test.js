const data = require('./__fixtures__/age-distrib-germany.json');

const { SORT_METHODS, getSymbolNormal } = require('../lib/const');

describe('sort methods', () => {
  describe('min', () => {
    it('should sort values from minimum to maximum', () => {
      expect(data.slice().sort(SORT_METHODS.min)).toMatchSnapshot();
    });
  });

  describe('max', () => {
    it('should sort values from maximum to minimum', () => {
      expect(data.slice().sort(SORT_METHODS.max)).toMatchSnapshot();
    });
  });

  describe('none', () => {
    it('should not sort anything', () => {
      expect(data.slice().sort(SORT_METHODS.none)).toEqual(data);
    });
  });
});

describe('getSymbolNormal', () => {
  const testValues = [
    -1,
    0,
    1 / 8,
    1 / 4,
    3 / 8,
    1 / 2,
    5 / 8,
    3 / 4,
    7 / 8,
    1,
    2,
  ];

  it('should return correct symbol for partial values', () => {
    expect(testValues.map(getSymbolNormal)).toMatchSnapshot();
  });
});
