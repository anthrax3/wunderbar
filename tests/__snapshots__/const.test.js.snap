// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSymbolNormal should return correct symbol for partial values 1`] = `
Array [
  "",
  "",
  "▏",
  "▎",
  "▍",
  "▌",
  "▋",
  "▊",
  "▉",
  "█",
  "█",
]
`;

exports[`sort methods max should sort values from maximum to minimum 1`] = `
Array [
  Object {
    "label": "0-14",
    "value": 10333117,
  },
  Object {
    "label": "15-24",
    "value": 8131788,
  },
  Object {
    "label": "25-54",
    "value": 32601939,
  },
  Object {
    "label": "55-64",
    "value": 11747501,
  },
  Object {
    "label": "65+",
    "value": 17779672,
  },
]
`;

exports[`sort methods min should sort values from minimum to maximum 1`] = `
Array [
  Object {
    "label": "0-14",
    "value": 10333117,
  },
  Object {
    "label": "15-24",
    "value": 8131788,
  },
  Object {
    "label": "25-54",
    "value": 32601939,
  },
  Object {
    "label": "55-64",
    "value": 11747501,
  },
  Object {
    "label": "65+",
    "value": 17779672,
  },
]
`;
